kubepy/definition_merger.py

```diff
@@ -11,10 +11,13 @@ def merge_definitions(*definitions):
 
 @merge_definitions.register(dict)
 def merge_dicts(*definitions):
-    keys = set(itertools.chain(*(definition.keys() for definition in definitions)))
+    if definitions[-1] is None:
+        return None
+    keys = set(itertools.chain(*(definition.keys() for definition in definitions if definition is not None)))
     result = {}
     for key in keys:
-        result[key] = merge_definitions(*(definition[key] for definition in definitions if key in definition))
+        result[key] = merge_definitions(*(definition[key] for definition in definitions
+                                          if definition is not None and key in definition))
```

Comment:
@glowka, as far as I see, in the current implementation key will not be unset. The `null` value will be set to the given key.
In order to unset key something like:
```suggestion
        value = merge_definitions(*(definition[key] for definition in definitions
                                          if definition is not None and key in definition))
        if value is not None:
            result[key] = value
```

should be written.

Reply:
The `null` value wasn't very problematic, but your solution is definitely cleaner. Done!